client/my-sites/checkout/src/components/payment-method-step.tsx

```diff
@@ -63,20 +64,22 @@ export default function PaymentMethodStep() {
 				<CheckoutTerms cart={ responseCart } />
 			</CheckoutTermsWrapper>
 
-			<WPOrderReviewSection>
-				<NonTotalPrices>
-					<NonProductLineItem subtotal lineItem={ getSubtotalLineItemFromCart( responseCart ) } />
-					{ taxLineItems.map( ( taxLineItem ) => (
-						<NonProductLineItem key={ taxLineItem.id } tax lineItem={ taxLineItem } />
-					) ) }
-					{ creditsLineItem && responseCart.sub_total_integer > 0 && (
-						<NonProductLineItem subtotal lineItem={ creditsLineItem } />
-					) }
-				</NonTotalPrices>
-				<TotalPrice>
-					<NonProductLineItem total lineItem={ getTotalLineItemFromCart( responseCart ) } />
-				</TotalPrice>
-			</WPOrderReviewSection>
+			{ ! hasCheckoutVersion( '2' ) && (
```

Comment:
Just thinking how to improve the readability of this, what about:

```
const RenderTaxLineItems = ({ taxLineItems }) => (
  <>
    {taxLineItems.map((taxLineItem) => (
      <NonProductLineItem key={taxLineItem.id} tax lineItem={taxLineItem} />
    ))}
  </>
);

const RenderCreditsLineItem = ({ creditsLineItem, responseCart }) => (
  <>
    {creditsLineItem && responseCart.sub_total_integer > 0 && (
      <NonProductLineItem subtotal lineItem={creditsLineItem} />
    )}
  </>
);

const RenderTotalPrice = ({ responseCart }) => (
  <TotalPrice>
    <NonProductLineItem total lineItem={getTotalLineItemFromCart(responseCart)} />
  </TotalPrice>
);

const OrderReview = ({ hasCheckoutVersion, responseCart, taxLineItems, creditsLineItem }) => {
  if (hasCheckoutVersion('2')) {
    return null; // Or some other component for version 2
  }

  return (
    <WPOrderReviewSection>
      <NonTotalPrices>
        <NonProductLineItem subtotal lineItem={getSubtotalLineItemFromCart(responseCart)} />
        <RenderTaxLineItems taxLineItems={taxLineItems} />
        <RenderCreditsLineItem creditsLineItem={creditsLineItem} responseCart={responseCart} />
      </NonTotalPrices>
      <RenderTotalPrice responseCart={responseCart} />
    </WPOrderReviewSection>
  );
};
```

Basically breaking things down and bringing them together.